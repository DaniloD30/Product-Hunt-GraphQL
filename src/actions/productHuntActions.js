import Constants from "../helpers/enums/Constants";

const savePosts = (popular, newest, callback = () => {}) => (dispatch) => {
  callback();
  dispatch({
    type: Constants.SAVE_POSTS,
    popular: popular,
    newest: newest,
  });
};

export default {
  savePosts
};
