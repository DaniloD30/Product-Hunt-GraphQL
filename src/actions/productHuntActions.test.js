import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import productHuntActions from "./productHuntActions";
export const mockStore = configureMockStore([thunk]);
const store = mockStore();

describe("actions should working to store", () => {
  beforeEach(() => {
    // Runs before each test in the suite
    store.clearActions();
  });

  describe("movieAction", () => {
    test("Dispatches the savePosts action and payload", () => {
      store.dispatch(productHuntActions.savePosts([], []));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
