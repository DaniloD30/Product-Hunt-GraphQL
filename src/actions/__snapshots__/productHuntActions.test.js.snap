// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should working to store movieAction Dispatches the savePosts action and payload 1`] = `
Array [
  Object {
    "newest": Array [],
    "popular": Array [],
    "type": "SAVE_POSTS",
  },
]
`;
